[PyTorch] Remove unnecessary dispatcher.h include in torch/library.h

swolchok opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#50317 [PyTorch] Remove unnecessary dispatcher.h include in torch/library.h**

It's unused.

Differential Revision: [D25859010](https://our.internmc.facebook.com/intern/diff/D25859010/)

swolchok committed:
[PyTorch] Remove unnecessary dispatcher.h include in torch/library.h

It's unused.

Differential Revision: [D25859010](https://our.internmc.facebook.com/intern/diff/D25859010/)

[ghstack-poisoned]

bhosmer commented:
not sure about the test failures though

bhosmer reviewed (commented):
At least some of these failures are from client code that was depending on definitions from dispatcher.h imported transitively through e.g. library.h - not sure if you want to add the necessary dispatcher.h includes directly to the clients or forget this one; should be simple to do the former

swolchok committed:
Update on "[PyTorch] Remove unnecessary dispatcher.h include in torch/library.h"


It's unused.

Differential Revision: [D25859010](https://our.internmc.facebook.com/intern/diff/D25859010/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Remove unnecessary dispatcher.h include in torch/library.h"


It's unused.

Differential Revision: [D25859010](https://our.internmc.facebook.com/intern/diff/D25859010/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Remove unnecessary dispatcher.h include in torch/library.h"


It's unused.

Differential Revision: [D25859010](https://our.internmc.facebook.com/intern/diff/D25859010/)

[ghstack-poisoned]

smessmer reviewed (approved):
This could be bc breaking for oss users who include `torch/library.h` and now need to add other includes. Likely they won't access the dispatcher directly, but there might be things that were transitively included from Dispatcher.h that they were using.
